var/spack/repos/builtin/packages/gaussian-src/package.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+import llnl.util.tty as tty
+import os
+import glob
+
+
+class GaussianSrc(Package):
+    """Gaussian is a computer program for computational chemistry.
+
+    This Spack package builds Gaussian from source.
+
+    Needs post-install steps to make it run!
+    See package installation log for details."""
+
+    homepage = "http://www.gaussian.com/"
+    manual_download = True
+
+    maintainers = ['dev-zero']
+
+    version('16-C.01', sha256='c9eb73a9df5ca8705fcf2d7ce2d5f9aceb05ae663689f54c0a581c9d4d44fffb')
+
+    depends_on('tcsh', type='build')
+
+    # All compilers except for pgi are in conflict:
+    for __compiler in spack.compilers.supported_compilers():
+        if __compiler != 'pgi':
+            conflicts('%{0}'.format(__compiler),
+                      msg='Gaussian can only be built with the PGI compiler')
+
+    patch('16-C.01-replace-deprecated-pgf77-with-pgfortran.patch', when='@16-C.01')
+    patch('16-C.01-fix-building-c-code-with-pgcc.patch', when='@16-C.01')
+    patch('16-C.01-fix-shebangs.patch', when='@16-C.01')
+
+    @property
+    def g_name(self):
+        return 'g{0}'.format(self.version.up_to(1))
+
+    @property
+    def g_root(self):
+        return self.prefix.join(self.g_name)
+
+    def url_for_version(self, version):
+        return "file://{0}/g{1}.tgz".format(os.getcwd(), version)
+
+    def install(self, spec, prefix):
+        # Spacks strips the single dir inside the tarball, but Gaussian
+        # needs it -> move them back
+        files = os.listdir()
+        mkdirp(self.g_name)
+        for f in files:
+            os.rename(f, join_path(self.g_name, f))
+
+        opts = ['all']
+        #  if spec.satisfies('+cuda'):
```

Comment:
Well, it would theoretically work but encountered several problems which I don't have time to fix:

* compilation is broken with pgi 20.4 due to Gaussian using deprecated ACC directives, while I fixed them I am not sure whether I can include the patch due to licensing restrictions
* compilation is broken with pgi 20.4 since Gaussian explicitly sets the CUDA version to use, but 20.4 does not include those CUDA versions anymore (can be remedied by patching out the pinned CUDA versions in `set-mflags`)
* compilation with the officially supported 18.10 is broken with a too recent system GCC: the PGI compilers seem to use the systems GCC's stdlibc++ headers which fail to build since that version of PGI does not support C++11/14
* btw, Spack's installation of `pgi@18.10~mpi` is broken

I would like to leave this comment in to show future devs how it could be enabled.